### PR TITLE
Introduce FHMAX_ENKF_GFS to extending ensemble forecast capabilities

### PR DESCRIPTION
**Pull request by XianwuXue-NOAA**
**Description**
Introduce FHMAX_ENKF_GFS to extending ensemble forecast capabilities for gfsefmn jobs

**Type of change**

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**
Tested it on WCOSS2

  
**Checklist**

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: scripts/exgdas_enkf_fcst.sh [hunk 1]
```diff
@@ -112,6 +112,12 @@ export FHOUT=${FHOUT_ENKF:-3}
 # model_configure
 export DELTIM=${DELTIM_ENKF:-${DELTIM:-225}}
 export FHMAX=${FHMAX_ENKF:-9}
+if [[ $EUPD_CYC == "BOTH" ]]; then
+    if [[ $CDUMP == "gfs" ]]; then
+        export FHMAXF=$FHMAX_ENKF_GFS
+    fi
+fi
```
Comment (aerorahul) [inline scripts/exgdas_enkf_fcst.sh]:
```suggestion
  if [[ ${CDUMP} == "gfs" ]]; then
      export FHMAX=${FHMAX_ENKF_GFS:-${FHMAX_ENKF:-9}}
  fi
```

File: workflow/rocoto/workflow_tasks.py [hunk 2]
```diff
@@ -1192,8 +1192,13 @@ def echgres(self):
     def epos(self):
 
         def _get_eposgroups(epos):
+            eupd_cyc=self._base['EUPD_CYC'].lower()
+
             fhmin = epos['FHMIN_ENKF']
             fhmax = epos['FHMAX_ENKF']
+            if eupd_cyc == "both":
+                if self.cdump == "gfs":
+                    fhmax = epos['FHMAX_ENKF_GFS']
```
Comment (aerorahul) [inline workflow/rocoto/workflow_tasks.py]:
```suggestion
              if self.cdump == "gfs":
                  fhmax = epos['FHMAX_ENKF_GFS']
```

**Review by aerorahul — CHANGES_REQUESTED** [review]
looks good.
some suggestions to remove superfluous if-blocks.

**Comment by XianwuXue-NOAA** [comment]
@aerorahul Thanks. Delete the if conditions.

File: scripts/exgdas_enkf_fcst.sh [hunk 3]
```diff
@@ -112,6 +112,10 @@ export FHOUT=${FHOUT_ENKF:-3}
 # model_configure
 export DELTIM=${DELTIM_ENKF:-${DELTIM:-225}}
 export FHMAX=${FHMAX_ENKF:-9}
+if [[ $CDUMP == "gfs" ]]; then
+   export FHMAXF=$FHMAX_ENKF_GFS
```
Comment (aerorahul) [inline scripts/exgdas_enkf_fcst.sh]: There is a typo here on the LHS.
Also, suggest this:
```suggestion
   export FHMAX=${FHMAX_ENKF_GFS:-${FHMAX_ENKF:-${FHMAX}}}
```

**Review by aerorahul — APPROVED** [review]
looks good.

**Pull request merged**